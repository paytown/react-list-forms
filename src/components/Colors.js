import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  const colorList = colors.map(color => {
    return (
      <li style={{ listStyle: 'none' }} key={color.name}>
        <Color name={color.name} hex={color.hex} />
      </li>
    );
  });

  return <ul style={{
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around'
  }}>{colorList}</ul>;
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;
