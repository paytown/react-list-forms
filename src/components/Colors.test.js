import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('color list', () => {
  it('renders colors from array', () => {
    const wrapper = shallow(<Colors colors={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
