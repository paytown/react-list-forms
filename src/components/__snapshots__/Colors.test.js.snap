// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color list renders colors from array 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
/>
`;
